### DISCUSSION TITLE
Currency format in ChartJS

### DISCUSSION THREAD

**Discussion by Read2uk** [Laravel-Backpack/community-forum#1219] [Q&A (Help)]
Hi all,

I am trying to build a few sales reports with ChartJS widgets but I am struggling to format the values;
If I format the figures in MySQL to have the correct format, they won't display in chartjs.

For example number formats I am dealing with: 14514313.31  , 17519321.73

How can I format to be in readable currency format? Do you guys have an example please?

There is documentation on how to do it if chartjs was in Javascript but not finding any documentation on php.

**Comment by karandatwani92** [comment]
Hey @Read2uk 

Let's say you want to display:
- 14514313.31 as 145K
- 17519321.73 as 175K

However, **the raw data point has to be a number**, not a string(145K). 
When we convert it into a string, ChartJS does not consider it and hides it.

I understand where you want it, but the scope that is possible right now is:
- In the chart widget header title, you can mention that the values are in `'header' => 'Sales(Thousands)'`.
- And.. turn them in thousands but keep them number type only:
```php
$this->chart->dataset('Sales', 'bar', [
            $sales_6_days_ago/1000,
            $sales_5_days_ago/1000,
            $sales_4_days_ago/1000,
            $sales_3_days_ago/1000,
            $sales_2_days_ago/1000,
            $yesterday_sales/1000,
            $today_sales/1000,
        ])->color('rgb(66, 186, 150, 1)')
            ->backgroundColor('rgb(66, 186, 150, 0.4)');
```
**Which will output:**:
![Screenshot 2024-11-06 at 6 44 13 PM](https://github.com/user-attachments/assets/9feb3d89-b076-4f85-873e-8731147cdad2)
